gm4_crossbow_cartridges/data/gm4_crossbow_cartridges/functions/check_projectile.mcfunction

```diff
@@ -16,7 +16,10 @@ execute if entity @s[nbt={Inventory:[{Slot:-106b,id:"minecraft:splash_potion"}]}
 execute if entity @s[nbt={Inventory:[{Slot:-106b,id:"minecraft:lingering_potion"}]}] run function gm4_crossbow_cartridges:projectile/lingerpotion
 
 #torch
-execute if entity @s[nbt={Inventory:[{Slot:-106b,id:"minecraft:torch"}]}] run function gm4_crossbow_cartridges:projectile/torch
+execute if entity @s[gamemode=!adventure,nbt={Inventory:[{Slot:-106b,id:"minecraft:torch"}]}] run function gm4_crossbow_cartridges:projectile/torch
```

Comment:
While you're updating this, it would make sense to move all these offhand checks to predicates